cmake/ElektraCompiling.cmake

```diff
@@ -61,10 +61,27 @@ if (CMAKE_CXX_COMPILER_ID MATCHES "Clang")
 
 	if (ENABLE_DEBUG)
 		set (EXTRA_FLAGS "${EXTRA_FLAGS} -fsanitize=undefined -fsanitize=integer")
-		set (CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -fsanitize=undefined -fsanitize=integer")
+		set (CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -lubsan")
 	endif()
 endif()
 
+if (ENABLE_ASAN)
+	set (EXTRA_FLAGS "${EXTRA_FLAGS} -fsanitize=address -fno-omit-frame-pointer")
+	set (ASAN_LIBRARY "-lasan") #this is needed for GIR to put asan in front
+	if (CMAKE_COMPILER_IS_GNUCXX)
+		set (CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -fsanitize=address")
+		# this is currently a workaround so gtests compile with ASAN
+		#TODO someone with expertise migrate this into LibAddTest
+		set (COMMON_FLAGS "${COMMON_FLAGS} -lpthread")
```

Comment:
Exactly, it *will* be needed ;) The *not* needed referred to the global pthread linker flag ;) Let us use #160 in future to avoid such confusions.

It would also be interesting to have it before the PR is merged: we can trigger the build job from here.